packages/resolver/src/adapters/cns-cardano-api-maker.ts

```diff
@@ -0,0 +1,121 @@
+import {CNSMetadata, CNSUserRecord} from './cns-types'
+import {hexToString} from './cns-utils'
+import {FetchData, fetchData, handleApiError, isLeft} from '@yoroi/common'
+import {AxiosRequestConfig} from 'axios'
+import {z} from 'zod'
+
+export const makeCnsCardanoApi = (
+  baseUrl: string,
+  fetcherConfig?: AxiosRequestConfig,
+  request: FetchData = fetchData,
+) => {
+  const getAssetAddress = async (
+    assetHex: string,
+  ): Promise<string | undefined> => {
+    const policyId = assetHex.slice(0, 56)
+    const assetName = assetHex.slice(56)
+
+    const response = await request<string>(
+      {
+        url: `${baseUrl}/api/asset/accounts?policy=${policyId}&asset=${assetName}`,
+      },
+      fetcherConfig,
+    )
+
+    if (isLeft(response)) {
+      handleApiError(response.error)
+    } else {
+      const parsedResponse = getAssetAddressSchema.parse(response.value.data)
+      return parsedResponse[0]
+    }
+  }
+
+  const getMetadata = async (
+    policyID: string,
+    assetName: string,
+  ): Promise<CNSMetadata | null> => {
+    const assetNameString = hexToString(assetName)
+    const key = `${policyID}.${assetNameString}`
+
+    const response = await request<CNSMetadata | null>(
+      {
+        url: `${baseUrl}/api/multiAsset/metadata`,
+        method: 'post',
+        data: {assets: [{nameHex: assetName, policy: policyID}]},
+      },
+      fetcherConfig,
+    )
+
+    if (isLeft(response)) {
+      handleApiError(response.error)
+    } else {
+      const parsedResponse = getMetadataSchema.parse(response.value.data)
+
+      if (Object.keys(parsedResponse).length === 0) return null
+
+      // @ts-ignore
+      return parsedResponse[key][0].metadata[policyID][assetNameString] ?? null
```

Comment:
Do you have any suggestion to fix this?

Reply:
<img width="544" alt="image" src="https://github.com/Emurgo/yoroi/assets/105349292/766e49ac-0524-48a1-ae68-01c6fead57b5">

Reply:
https://github.com/Emurgo/yoroi/blob/develop/packages/api/src/cardano/api/token-onchain-metadata.ts#L13